spec/unit/outputs/elasticsearch/http_client/pool_spec.rb

```diff
@@ -235,4 +240,31 @@
       end
     end
   end
+
+  describe "license checking" do
+    before(:each) do
+      allow(subject).to receive(:health_check_request)
+    end
+    context "when using default logstash distribution" do
+      let(:oss) { false }
+      context "if ES doesn't return a valid license" do
+        let(:valid_license) { false }
+        it "marks the url as active" do
+          subject.update_initial_urls
+          expect(subject.alive_urls_count).to eq(1)
+        end
+        it "logs a warning" do
+          expect(subject).to receive(:log_license_deprecation_warn).once
+          subject.update_initial_urls
+        end
+      end
+      context "if ES returns a valid license" do
+        let(:valid_license) { true }
+        it "marks the url as active" do
+          subject.update_initial_urls
+          expect(subject.alive_urls_count).to eq(1)
+        end
+      end
```

Comment:
Does it make sense to have a test proving the license deprecation is not logged in the license is there?